lib/slack.ml

```diff
@@ -453,28 +453,41 @@ let generate_status_notification ~(job_log : (string * string) list) ~(cfg : Con
     ~channel:(Status_notification.to_slack_channel channel)
     ()
 
-let generate_failed_build_notification ?slack_user_id ?(is_fix_build_notification = false) ~failed_steps
-  (n : Util.Webhook.n) channel =
+let generate_failed_build_notification ?(slack_ids : user_id option list = []) ?(is_fix_build_notification = false)
+  ~(cfg : Config_t.config) ~failed_steps (n : Util.Webhook.n) channel =
   let pipeline_name = Util.Webhook.pipeline_name n in
   let repo_url = Util.Build.git_ssh_to_https n.pipeline.repository in
   let commit_url = sprintf "%s/commit/%s" repo_url n.build.sha in
   let summary =
+    (* check if this is an escalation notification. if it's mixed, paciencia *)
+    let is_escalation_notification =
+      Util.Webhook.get_escalation_threshold cfg n
+      |> Option.map_default
+           (fun escalation_threshold ->
+             FailedStepSet.exists
+               (fun step ->
+                 (* escalate steps that have been broken for longer than the threshold *)
+                 Util.Webhook.is_past_span step.created_at escalation_threshold)
+               failed_steps)
+           false
```

Comment:
```suggestion
      match Util.Webhook.get_escalation_threshold cfg n with
      | None -> false
      | Some escalation_threshold ->
             FailedStepSet.exists
               (fun step ->
                 (* escalate steps that have been broken for longer than the threshold *)
                 Util.Webhook.is_past_span step.created_at escalation_threshold)
               failed_steps
```

a bit simpler to write it plainer

Reply:
i'm always on the fence on one or the other.. I thought about it, and decided for the Option version, but I think it's good to go with the match option. Thanks